cli/src/commands/extensions/api.rs

```diff
@@ -0,0 +1,256 @@
+                (p.id, p.group_name)
+            } else {
+                return Err(anyhow!("Failed to find a valid project configuration"));
+            }
+        }
+    };
+
+    let job_id = api
+        .submit_request(
+            &request_type,
+            &packages,
+            false,
+            project,
+            None,
+            group.map(String::from),
+        )
+        .await?;
+
+    Ok(job_id)
+}
+
+/// Retrieve user info.
+/// Equivalent to `phylum auth status`.
+#[op]
+async fn get_user_info(state: Rc<RefCell<OpState>>) -> Result<UserInfo> {
+    let api = ExtensionStateRef::from(state).await?;
+
+    api.user_info().await.map_err(Error::from)
+}
+
+/// Retrieve the access token.
+/// Equivalent to `phylum auth token --bearer`.
+#[op]
+async fn get_access_token(state: Rc<RefCell<OpState>>, ignore_certs: bool) -> Result<AccessToken> {
+    let refresh_token = get_refresh_token::call(state.clone()).await?;
+    let api = ExtensionStateRef::from(state).await?;
+    let config = api.config();
+
+    let access_token =
+        crate::auth::handle_refresh_tokens(&refresh_token, ignore_certs, &config.connection.uri)
+            .await?
+            .access_token;
+    Ok(access_token)
+}
+
+/// Retrieve the refresh token.
+/// Equivalent to `phylum auth token`.
+#[op]
+async fn get_refresh_token(state: Rc<RefCell<OpState>>) -> Result<RefreshToken> {
+    let api = ExtensionStateRef::from(state).await?;
+    let config = api.config();
+
+    config
+        .auth_info
+        .offline_access
+        .clone()
+        .ok_or_else(|| anyhow!("User is not currently authenticated"))
+}
+
+/// Retrieve a job's status.
+/// Equivalent to `phylum history job`.
+#[op]
+async fn get_job_status(
+    state: Rc<RefCell<OpState>>,
+    job_id: Option<&str>,
+) -> Result<JobStatusResponse<PackageStatusExtended>> {
+    let api = ExtensionStateRef::from(state).await?;
+
+    let job_id = job_id
+        .map(|job_id| JobId::from_str(job_id).ok())
+        .unwrap_or_else(|| get_current_project().map(|p| p.id))
+        .ok_or_else(|| anyhow!("Failed to find a valid project configuration"))?;
+    api.get_job_status_ext(&job_id).await.map_err(Error::from)
+}
+
+/// Retrieve a project's details.
+/// Equivalent to `phylum history project`.
+#[op]
+async fn get_project_details(
+    state: Rc<RefCell<OpState>>,
+    project_name: Option<&str>,
+) -> Result<ProjectDetailsResponse> {
+    let api = ExtensionStateRef::from(state).await?;
+
+    let project_name = project_name
+        .map(String::from)
+        .map(Result::Ok)
+        .unwrap_or_else(|| {
+            get_current_project()
+                .map(|p| p.name)
+                .ok_or_else(|| anyhow!("Failed to find a valid project configuration"))
+        })?;
+    api.get_project_details(&project_name)
+        .await
+        .map_err(Error::from)
+}
+
+/// Analyze a single package.
+/// Equivalent to `phylum package`.
+#[op]
+async fn analyze_package(
```

Comment:
Calling this `analyze_package()` makes it sound like it will submit the package for analysis, which is not true. The `PhylumApi` function is `get_package_details()` and I think that name (or a variant of it) better matches the actual functionality.

Reply:
Renamed this in the latest commit.